### PR TITLE
chore: add timeouts to github action jobs

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

On occasion we run out of github actions runners which slows people down.

The last two times this has happened there have been multiple jobs running for longer than an hour

## Changes

adds shorter timeouts to some likely candidates

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

opening the PR to see what GH does with the settings

### CODE REVIEW

File: .github/workflows/ci-backend.yml [hunk 1]
```diff
@@ -167,6 +171,7 @@ jobs:
 
     django:
         needs: changes
+        timeout-minutes: 30
```
Comment (pauldambra) [inline .github/workflows/ci-backend.yml:174]:
```suggestion
        timeout-minutes: 60
```

File: .github/workflows/ci-backend.yml [hunk 2]
```diff
@@ -270,6 +275,7 @@ jobs:
 
     cloud:
         needs: changes
+        timeout-minutes: 30
```
Comment (pauldambra) [inline .github/workflows/ci-backend.yml:278]:
```suggestion
        timeout-minutes: 60
```

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -28,6 +28,7 @@ jobs:
     cypress_prep:
         name: Cypress E2E preparation
         runs-on: ubuntu-latest
+        timeout-minutes: 30
```
Comment (pauldambra) [inline .github/workflows/e2e.yml:31]:
```suggestion
        timeout-minutes: 60
```

File: .github/workflows/e2e.yml [hunk 4]
```diff
@@ -57,6 +58,7 @@ jobs:
         name: Cypress E2E tests (${{ strategy.job-index }})
         if: ${{ github.ref != 'refs/heads/master' }} # Don't run on master, we only cace about node_modules cache
         runs-on: ubuntu-18.04
+        timeout-minutes: 30
```
Comment (pauldambra) [inline .github/workflows/e2e.yml:61]:
```suggestion
        timeout-minutes: 60
```

**Comment by EDsCODE** [comment]
Do they even need to be 60 minutes?

**Comment by pauldambra** [comment]
> Do they even need to be 60 minutes?

Yeah, I set them to 30 minutes initially but I just saw an e2e test run that spent 16 minutes downloading its cache. So added some headroom

I guess its better not to fail if there's nothing we could have done about the failure but 🤷

**Review by EDsCODE — APPROVED** [review]

**Comment by pauldambra** [comment]
@EDsCODE looking over the last 5 days we would have had very few timeouts at 30 minutes

So I've flipped back to that.

**Pull request merged**